pkg/pluginsupplier/plugin_supplier.go

```diff
@@ -48,6 +49,31 @@ func GetInstalledServerPlugins() ([]cli.PluginInfo, error) {
 	return serverPlugins, nil
 }
 
+// FilterPluginsByActiveContextType will exclude any plugin with an explicit
+// setting of supportedContextType that does not match the type of any active CLI context
+// Separating this conditional check so GetInstalledStandalonePlugins can
+// continue to return all standalone plugins regardless of supportedContextType
+func FilterPluginsByActiveContextType(plugins []cli.PluginInfo) (result []cli.PluginInfo, err error) {
```

Comment:
I didn't want to the potential filtering via active contexts factor into existing plugin-listing functions in this package, hence introducing it as another function. Curious if others have suggestions to alternatives? 
cc @marckhouzam @anujc25

Reply:
I think this seems a reasonable thing to start with. We can enhance and update this once we finalize the USX around listing plugins with the `tanzu plugin list` command.